go/nbs/table_reader.go

```diff
@@ -345,6 +301,72 @@ func (tr tableReader) getMany(reqs []getRecord) (remaining bool) {
 	return
 }
 
+// findOffsets iterates over |reqs| and |tr.prefixes| (both sorted by
+// address) to build the set of table locations which must be read in order to
+// find each chunk specified by |reqs|. If this table contains all requested
+// chunks remaining will be set to false upon return. If some are not here,
+// then remaining will be true.
+func (tr tableReader) findOffsets(reqs []getRecord) (ors offsetRecSlice, remaining bool) {
```

Comment:
`findOffsets()` and `canReadAhead()` are intended to have the same semantics as the code they were refactored from, so please check me and make sure I didn't change some edge case that's not covered by tests or something